src/ConfigTemplateDefinition.php

```diff
@@ -100,17 +112,21 @@ public static function createFromFile($file_name) {
    * @param string $key
    *   The key for template.
    *
-   * @return array|string
+   * @return mixed
    *   Returns value of base template definition property.
    */
   public function getKey($key) {
+    if (!isset($this->definition[$key])) {
+      return NULL;
```

Comment:
> not all places, where this is called are handling the NULL correctly. e.g in class ConfigTemplateDefinitionCollection it is assumed, that this is always an array. also a couple of times in class ConfigTemplateDefinitionResolver.

I fixed handling NULL values by checking the return value in aforementioned places.

> Also, is it correct, that this can return string or array? This is a bit strange.

It looks like this is just returning values from [testsite_builder.config_template.views_node_content.yml](https://github.com/thunder/testsite_builder/blob/feat/refactoring-of-config-template-importer/template/testsite_builder.config_template.views_node_content.yml) that might be an array or a string.